File: app/src/main/java/com/tobiasfried/brewkeeper/GenericPickerDialog.java
package com.tobiasfried.brewkeeper;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;

import java.util.Objects;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

public class GenericPickerDialog extends DialogFragment {

    // Member Variables
    private static String mResourceName = "arrayResourceId";
    private DialogInterface.OnClickListener mListener;

    // Instead of generic constructor
    public static GenericPickerDialog newInstance(int arrayResource) {
        GenericPickerDialog fragment = new GenericPickerDialog();
        Bundle args = new Bundle();
        args.putInt(mResourceName, arrayResource);
        fragment.setArguments(args);
        return fragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(Objects.requireNonNull(getActivity()));
        builder.setItems(getArguments().getInt(mResourceName), mListener);
        return builder.create();
    }

    public void setOnClickListener(DialogInterface.OnClickListener listener) {
        this.mListener = listener;
    }

}

File: app/src/main/java/com/tobiasfried/rasa/GenericPickerDialog.kt
package com.tobiasfried.rasa

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle

import java.util.Objects
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

class GenericPickerDialog : DialogFragment() {
    private var mListener: DialogInterface.OnClickListener? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(Objects.requireNonNull<FragmentActivity>(activity))
        builder.setItems(arguments!!.getInt(mResourceName), mListener)
        return builder.create()
    }

    fun setOnClickListener(listener: DialogInterface.OnClickListener) {
        this.mListener = listener
    }

    companion object {

        // Member Variables
        private val mResourceName = "arrayResourceId"

        // Instead of generic constructor
        fun newInstance(arrayResource: Int): GenericPickerDialog {
            val fragment = GenericPickerDialog()
            val args = Bundle()
            args.putInt(mResourceName, arrayResource)
            fragment.arguments = args
            return fragment
        }
    }

}
